tests/catalog/test_glue.py

```diff
@@ -0,0 +1,493 @@
+    from mypy_boto3_glue.type_defs import TableTypeDef as GlueTableInfo
+
+    from daft.dataframe import DataFrame
+else:
+    GlueClient = Any
+    GlueColumnInfo = Any
+    GlueTableInfo = Any
+
+
+@pytest.fixture
+def mock_session():
+    with mock_aws():
+        yield boto3.Session(region_name="us-west-2")
+
+
+@pytest.fixture
+def glue_client():
+    with mock_aws():
+        yield boto3.client("glue", region_name="us-west-2")
+
+
+@pytest.fixture
+def glue_catalog(glue_client):
+    gc = GlueCatalog.from_client("mock_glue_catalog", glue_client)
+    gc._table_impls.append(GlueTestTable)  # !! REGISTER GLUE TEST TABLE !!
+    return gc
+
+
+###
+# catalog constructors
+###
+
+
+def test_load_glue():
+    catalog = load_glue(
+        "mock_glue_catalog",
+        region_name="us-west-2",
+        api_version="2017-03-31",
+        use_ssl=False,
+        verify=True,
+        endpoint_url="http://localhost",
+        aws_access_key_id="test_access_key",
+        aws_secret_access_key="test_secret_key",
+        aws_session_token="test_session_token",
+    )
+    assert catalog.name == "mock_glue_catalog"
+
+
+def test_catalog_from_client(glue_client):
+    assert Catalog.from_glue("gc", client=glue_client)
+    assert GlueCatalog.from_client("gc", glue_client)
+
+
+def test_catalog_from_session(mock_session):
+    assert Catalog.from_glue("gc", session=mock_session)
+    assert GlueCatalog.from_session("gc", session=mock_session)
+
+
+def test_catalog_no_args():
+    with pytest.raises(ValueError, match="Must provide either a client or session."):
+        Catalog.from_glue("foo")
+
+
+def test_glue_catalog_init():
+    with pytest.raises(ValueError, match="not supported"):
+        GlueCatalog()
+
+
+###
+# catalog methods
+###
+
+
+def test_create_namespace(glue_catalog, glue_client):
+    # create a namespace with catalog
+    glue_catalog.create_namespace("test_namespace")
+
+    # verify with client
+    res = glue_client.get_databases()
+    assert len(res["DatabaseList"]) == 1
+    assert res["DatabaseList"][0]["Name"] == "test_namespace"
+
+    # err. already exists
+    with pytest.raises(ValueError, match="already exists"):
+        glue_catalog.create_namespace("test_namespace")
+
+
+def test_create_namespace_if_not_exists(glue_catalog, glue_client):
+    # create a namespace with catalog
+    glue_catalog.create_namespace_if_not_exists("test_namespace")
+    glue_catalog.create_namespace_if_not_exists("test_namespace")
+
+    # verify we only have one with client
+    res = glue_client.get_databases()
+    assert len(res["DatabaseList"]) == 1
+
+
+def test_drop_namespace(glue_catalog, glue_client):
+    # create then drop the namespace
+    glue_catalog.create_namespace("test_namespace")
+    glue_catalog.drop_namespace("test_namespace")
+
+    # verify was dropped with client
+    res = glue_client.get_databases()
+    assert len(res["DatabaseList"]) == 0
+
+    # drop if doesn"t exist should error
```

Comment:
lol, uber nit, just spotted it
```suggestion
    # drop if doesn't exist should error
```

Reply:
All good lol, I notice I've made a lot of typos in docs. Poor stream of consciousness writing style..